Example with Port. An error with my test

could you provides an example using port monitoring ?
I try : 

```
resource "uptimerobot_monitor" "kubernetes_master" {
  url           = "tcp:/my.example.com:5555"
  type          = "port"
  sub_type      = "custom"
  port          = "5555"
  friendly_name = "An example"
}
```

```
resource "uptimerobot_monitor" "kubernetes_master" {
  url           = "my.example.com"
  type          = "port"
  sub_type      = "custom"
  port          = "5555"
  friendly_name = "An example"
}
```

I've got this error : 

```
Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


panic: interface conversion: interface {} is string, not int
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: 
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: goroutine 32 [running]:
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: github.com/louy/terraform-provider-uptimerobot/uptimerobot.resourceMonitorCreate(0xc0001edf10, 0xe0d960, 0xc0004b6e60, 0xc0001edf10, 0x0)
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: 	/home/nlamirault/Projects/terraform-provider-uptimerobot/uptimerobot/resource_uptimerobot_monitor.go:115 +0xa25
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc0001ed1f0, 0xc00038c6e0, 0xc00001d320, 0xe0d960, 0xc0004b6e60, 0x40d601, 0xc000519b80, 0x4c3d7c)
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: 	/home/nlamirault/Applications/golang/pkg/mod/github.com/hashicorp/terraform@v0.11.7/helper/schema/resource.go:227 +0x351
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc0001ed2d0, 0xc00038c690, 0xc00038c6e0, 0xc00001d320, 0xc00005aa80, 0x18, 0x7f0cce5b86c0)
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: 	/home/nlamirault/Applications/golang/pkg/mod/github.com/hashicorp/terraform@v0.11.7/helper/schema/provider.go:283 +0x9c
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Apply(0xc00001c640, 0xc00001cee0, 0xc0002c9480, 0x0, 0x0)
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: 	/home/nlamirault/Applications/golang/pkg/mod/github.com/hashicorp/terraform@v0.11.7/plugin/resource_provider.go:527 +0x57
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: reflect.Value.call(0xc0003844e0, 0xc0001746f8, 0x13, 0xe4354b, 0x4, 0xc000519f18, 0x3, 0x3, 0xc000181980, 0x414fe7, ...)
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: 	/home/nlamirault/Applications/go/src/reflect/value.go:447 +0x454
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: reflect.Value.Call(0xc0003844e0, 0xc0001746f8, 0x13, 0xc000365f18, 0x3, 0x3, 0x12a05f200, 0xc000365f10, 0xc000365fb8)
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: 	/home/nlamirault/Applications/go/src/reflect/value.go:308 +0xa4
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: net/rpc.(*service).call(0xc00049a800, 0xc00038c4b0, 0xc000212cf8, 0xc000212d10, 0xc00039c580, 0xc000438580, 0xcacd40, 0xc00001cee0, 0x16, 0xcacd80, ...)
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: 	/home/nlamirault/Applications/go/src/net/rpc/server.go:384 +0x14e
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: created by net/rpc.(*Server).ServeCodec
2019-02-06T09:44:15.866+0100 [DEBUG] plugin.terraform-provider-uptimerobot: 	/home/nlamirault/Applications/go/src/net/rpc/server.go:481 +0x47e
2019-02-06T09:44:15.867+0100 [DEBUG] plugin: plugin process exited: path=/home/nlamirault/.terraform.d/plugins/linux_amd64/terraform-provider-uptimerobot
2019/02/06 09:44:15 [TRACE] root: eval: *terraform.EvalWriteState
2019/02/06 09:44:15 [TRACE] root: eval: *terraform.EvalApplyProvisioners
2019/02/06 09:44:15 [TRACE] root: eval: *terraform.EvalIf
2019/02/06 09:44:15 [TRACE] root: eval: *terraform.EvalWriteState
2019/02/06 09:44:15 [TRACE] root: eval: *terraform.EvalWriteDiff
2019/02/06 09:44:15 [TRACE] root: eval: *terraform.EvalApplyPost
2019/02/06 09:44:15 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* uptimerobot_monitor.kubernetes_master: unexpected EOF
2019/02/06 09:44:15 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* uptimerobot_monitor.kubernetes_master: unexpected EOF
2019/02/06 09:44:15 [TRACE] [walkApply] Exiting eval tree: uptimerobot_monitor.kubernetes_master
2019/02/06 09:44:15 [TRACE] dag/walk: upstream errored, not walking "meta.count-boundary (count boundary fixup)"
2019/02/06 09:44:15 [TRACE] dag/walk: upstream errored, not walking "provider.uptimerobot (close)"
2019/02/06 09:44:15 [TRACE] dag/walk: upstream errored, not walking "root"
2019/02/06 09:44:15 [TRACE] Preserving existing state lineage "04a4a4b1-6432-89b5-083f-72a0cf33f934"
2019/02/06 09:44:15 [TRACE] Preserving existing state lineage "04a4a4b1-6432-89b5-083f-72a0cf33f934"
2019/02/06 09:44:15 [TRACE] Preserving existing state lineage "04a4a4b1-6432-89b5-083f-72a0cf33f934"
2019/02/06 09:44:15 [TRACE] Preserving existing state lineage "04a4a4b1-6432-89b5-083f-72a0cf33f934"
2019/02/06 09:44:15 [DEBUG] plugin: waiting for all plugin processes to complete...

```

--- a/uptimerobot/api/monitor.go
+++ b/uptimerobot/api/monitor.go
@@ -96,7 +96,11 @@ func (client UptimeRobotApiClient) GetMonitor(id int) (m Monitor, err error) {
 	switch m.Type {
 	case "port":
 		m.SubType = intToString(monitorSubType, int(monitor["sub_type"].(float64)))
-		m.Port = int(monitor["port"].(float64))
+		if (m.SubType != "custom") {
+			m.Port = 0
+		} else {
+			m.Port = int(monitor["port"].(float64))
+		}
 		break
 	case "keyword":
 		m.KeywordType = intToString(monitorKeywordType, int(monitor["keyword_type"].(float64)))

--- a/uptimerobot/resource_uptimerobot_monitor.go
+++ b/uptimerobot/resource_uptimerobot_monitor.go
@@ -41,7 +41,7 @@ func resourceMonitor() *schema.Resource {
 				// required for port monitoring
 			},
 			"port": {
-				Type:     schema.TypeString,
+				Type:     schema.TypeInt,
 				Optional: true,
 				// required for port monitoring
 			},

--- a/uptimerobot/resource_uptimerobot_monitor_test.go
+++ b/uptimerobot/resource_uptimerobot_monitor_test.go
@@ -43,11 +43,11 @@ func TestUptimeRobotDataResourceMonitor_http_monitor(t *testing.T) {
 	})
 }
 
-func TestUptimeRobotDataResourceMonitor_port_monitor(t *testing.T) {
-	var FriendlyName = "TF Test: http monitor"
+func TestUptimeRobotDataResourceMonitor_http_port_monitor(t *testing.T) {
+	var FriendlyName = "TF Test: http port monitor"
 	var Type = "port"
 	var URL = "google.com"
-	var Port = "8080"
+	var SubType = "http"
 	resource.Test(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
@@ -59,14 +59,52 @@ func TestUptimeRobotDataResourceMonitor_port_monitor(t *testing.T) {
 					friendly_name = "%s"
 					type          = "%s"
 					url           = "%s"
-					port          = "%s"
+					sub_type      = "%s"
 				}
-				`, FriendlyName, Type, URL, Port),
+				`, FriendlyName, Type, URL, SubType),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "friendly_name", FriendlyName),
 					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "type", Type),
 					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "url", URL),
-					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "port", Port),
+					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "sub_type", SubType),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "uptimerobot_monitor.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestUptimeRobotDataResourceMonitor_custom_port_monitor(t *testing.T) {
+	var FriendlyName = "TF Test: custom port monitor"
+	var Type = "port"
+	var URL = "google.com"
+	var SubType = "custom"
+	var Port = 8080
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMonitorDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(`
+				resource "uptimerobot_monitor" "test" {
+					friendly_name = "%s"
+					type          = "%s"
+					url           = "%s"
+					sub_type      = "%s"
+					port          = %d
+				}
+				`, FriendlyName, Type, URL, SubType, Port),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "friendly_name", FriendlyName),
+					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "type", Type),
+					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "url", URL),
+					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "sub_type", SubType),
+					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "port", fmt.Sprintf(`%d`, Port)),
 				),
 			},
 			resource.TestStep{

--- a/uptimerobot/resource_uptimerobot_monitor_test.go
+++ b/uptimerobot/resource_uptimerobot_monitor_test.go
@@ -43,6 +43,41 @@ func TestUptimeRobotDataResourceMonitor_http_monitor(t *testing.T) {
 	})
 }
 
+func TestUptimeRobotDataResourceMonitor_port_monitor(t *testing.T) {
+	var FriendlyName = "TF Test: http monitor"
+	var Type = "port"
+	var URL = "google.com"
+	var Port = "8080"
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMonitorDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(`
+				resource "uptimerobot_monitor" "test" {
+					friendly_name = "%s"
+					type          = "%s"
+					url           = "%s"
+					port          = "%s"
+				}
+				`, FriendlyName, Type, URL, Port),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "friendly_name", FriendlyName),
+					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "type", Type),
+					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "url", URL),
+					resource.TestCheckResourceAttr("uptimerobot_monitor.test", "port", Port),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "uptimerobot_monitor.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
 func TestUptimeRobotDataResourceMonitor_change_url(t *testing.T) {
 	var FriendlyName = "TF Test: http monitor"
 	var Type = "http"

